cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -336,3 +335,66 @@ func TestConfigCertAndKey(t *testing.T) {
 		)
 	}
 }
+
+func TestKubernetesVersionCheck(t *testing.T) {
+	var tests = []struct {
+		check    KubernetesVersionCheck
+		expected bool
```

Comment:
Since the function signature supports returning both warnings and errors separately, I would expect to instead see `expectWarnings` and `expectErrors`, if we think there's an important enough distinction between them to model them in the API.